app/components/Sidebar/components/TrashLink.js

```diff
@@ -0,0 +1,61 @@
+// @flow
+import { observer } from "mobx-react";
+import { TrashIcon } from "outline-icons";
+import * as React from "react";
+import { useState } from "react";
+import { useDrop } from "react-dnd";
+import { useTranslation } from "react-i18next";
+import DocumentDelete from "scenes/DocumentDelete";
+import Modal from "components/Modal";
+import useStores from "../../../hooks/useStores";
+import SidebarLink from "./SidebarLink";
+
+function TrashLink({ documents }) {
+  const { policies } = useStores();
+  const { t } = useTranslation();
+  const [document, setDocument] = useState();
+
+  const [{ isDocumentDropping }, dropToTrashDocument] = useDrop({
+    accept: "document",
+    drop: async (item, monitor) => {
```

Comment:
```suggestion
    drop: (item, monitor) => {
```

Try removing? Unnecessary anyway 🤷 – a debugger in here is triggered or no?

Reply:
My best guess is that maybe it's opening the modal but then the following `mouseup` is counting as a click "outside" the modal and it immediately closes. If that's the case wrapping the setDocument in a `setTimeout` might do the trick

Reply:
Bull's-eye! It worked!!

Reply:
This is the issue because override the `hover` instead of `drop` does the work.

Reply:
Hurray, we did it 🎉